dev/gp-gcloud/cmd/compute/instance-templates-create.go

```diff
@@ -0,0 +1,300 @@
+		project              string
+		machineType          string
+		minCpuPlatform       string
+		serviceAccount       string
+		serviceAccountScopes string
+		bootDiskSize         string
+		bootDiskType         string
+		tags                 string
+		labels               string
+		image                string
+		networkInterface     string
+		metadata             string
+		metadataFromFile     string
+		localSSDs            []string
+		noRestartOnFailure   bool
+		dryRun               bool
+	)
+
+	cmd := &cobra.Command{
+		Use:     "create",
+		Short:   "create wrapper",
+		Example: "  create instance-template-1",
+		Run: func(cmd *cobra.Command, args []string) {
+			if len(args) != 1 {
+				log.Fatal("no name was provided")
+			}
+
+			name := args[0]
+
+			ctx := context.Background()
+
+			c, err := google.DefaultClient(ctx, compute.CloudPlatformScope)
+			if err != nil {
+				log.Fatal(err)
+			}
+
+			computeService, err := compute.New(c)
+			if err != nil {
+				log.Fatal(err)
+			}
+
+			networkInterfaceParsed := parseLabels(networkInterface)
+			automaticRestart := !noRestartOnFailure
+			metadata := parseMetadata(metadata)
+			metadata = append(metadata, parseMetadataFromFile(metadataFromFile)...)
+			localSSDDevices := parseLocalSSDs(localSSDs)
+			labels := parseLabels(labels)
+
+			rb := &compute.InstanceTemplate{
+				Name: name,
+				Properties: &compute.InstanceProperties{
+					MachineType:    machineType,
+					MinCpuPlatform: minCpuPlatform,
+					ServiceAccounts: []*compute.ServiceAccount{
+						{
+							Email:  serviceAccount,
+							Scopes: strings.Split(serviceAccountScopes, ","),
+						},
+					},
+					Disks: []*compute.AttachedDisk{
+						{
+							AutoDelete: true,
+							Boot:       true,
+							InitializeParams: &compute.AttachedDiskInitializeParams{
+								DiskSizeGb:  parseDiskSize(bootDiskSize, 10),
+								DiskType:    bootDiskType,
+								SourceImage: image,
+								Labels:      labels,
+							},
+						},
+					},
+					Tags: &compute.Tags{
+						Items: strings.Split(tags, ","),
+					},
+					Labels: labels,
+					NetworkInterfaces: []*compute.NetworkInterface{
+						{
+							AccessConfigs: []*compute.AccessConfig{
+								{
+									NetworkTier: networkInterfaceParsed["network-tier"],
+									Type:        "ONE_TO_ONE_NAT",
+								},
+							},
+							Network:    networkInterfaceParsed["network"],
+							Subnetwork: networkInterfaceParsed["subnet"],
+						},
+					},
+					Metadata: &compute.Metadata{
+						Items: metadata,
+					},
+					Scheduling: &compute.Scheduling{
+						AutomaticRestart: &automaticRestart,
+					},
+				},
+			}
+
+			for _, localSSD := range localSSDDevices {
+				rb.Properties.Disks = append(rb.Properties.Disks, &compute.AttachedDisk{
+					AutoDelete: true,
+					Boot:       false,
+					InitializeParams: &compute.AttachedDiskInitializeParams{
+						DiskType: "local-ssd",
+						Labels:   labels,
+					},
+					Type:      "SCRATCH",
```

Comment:
How do we know this is the right type?

Reply:
there are only two types. scratch and permanent. you cannot have local ssd with type permanent.